File: mvpcore/src/main/java/biz/dealnote/mvp/core/IPresenter.java
package biz.dealnote.mvp.core;

import android.os.Bundle;
import android.support.annotation.NonNull;

/**
 * Created by admin on 11.07.2016.
 * mvpcore
 */
public interface IPresenter<V extends IMvpView> {
    void saveState(@NonNull Bundle savedState);

    void destroy();
    void resumeView();
    void pauseView();
    void attachViewHost(@NonNull V view);
    void detachViewHost();
    void createView(@NonNull V view);
    void destroyView();
}

File: mvpcore/src/main/java/biz/dealnote/mvp/core/IPresenter.kt
package biz.dealnote.mvp.core

import android.os.Bundle

/**
 * Created by admin on 11.07.2016.
 * mvpcore
 */
interface IPresenter<V : IMvpView> {
    fun saveState(outState: Bundle)

    fun destroy()
    fun resumeView()
    fun pauseView()
    fun attachViewHost(view: V)
    fun detachViewHost()
    fun createView(view: V)
    fun destroyView()
}